chainer/initializers/__init__.py

```diff
@@ -65,16 +65,7 @@ def generate_array(initializer, shape, xp, dtype=None, device=None):
             raise ValueError('xp and device arguments are inconsistent.')
 
     if xp is chainerx:
-        # Initialize with NumPy/CuPy array that shares memory with the
-        # ChainerX array.
-        # TODO(sonots): Directly use initializer after ChainerX
-        # supports random.
-        chx_device = backend_device.device
-        array = chainerx.empty(shape, dtype=dtype, device=chx_device)
-        fallback_device = backend_device.fallback_device
-        with chainer.using_device(fallback_device):
-            initializer(fallback_device.send(array))
-        return array
+        backend_device = backend_device.device
```

Comment:
You can remove these two lines.
Calling `chainer.using_device` on `chainerx.Device` would create another `chainer.backend.ChainerxDevice` instance which is unnecessary overhead.

Reply:
Some of the tests (not the ones written by me) have the default device as cuda:0 or native:0 but sets the argument device to be native:1 or cuda:1, making the tests fail if `with chainer.using_device` is not used.

Is this expected behavior?

Reply:
Oh, I didn't mean removing `using_device`, but lines 67-68.

Reply:
Done! :D